### PR TITLE
A small fix to change pounds to pence before submitting it to products

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT
`publicapi` expects pence as payment amount

## HOW 
replace `.` with no space before posting it to products

### CODE REVIEW

**Comment by kakumara** [comment]
is the description correct. As far as I know selfservice does not interact with public api. I believe you mean products ???

File: app/controllers/make_a_demo_payment/confirm_controller.js [hunk 1]
```diff
@@ -15,6 +15,8 @@ module.exports = (req, res) => {
   let gatewayAccountId = auth.getCurrentGatewayAccountId(req)
 
   if (protoData) {
+    let paymentAmount = protoData.protoPaymentAmount.replace(new RegExp('\\.', 'g'), '')
```
Comment (kakumara) [inline app/controllers/make_a_demo_payment/confirm_controller.js]: Is this correct? Shouldn't you be multiplying the `protoData.protoPaymentAmount` * 100 ? 
Say if I type `10.000` it still means I expect £10 (1000 pence) not £100 (10000 pence)

Comment (SandorArpa) [reply app/controllers/make_a_demo_payment/confirm_controller.js]: it is checked in previous controller before sent to this one

File: app/controllers/make_a_demo_payment/confirm_controller.js [hunk 2]
```diff
@@ -15,6 +15,12 @@ module.exports = (req, res) => {
   let gatewayAccountId = auth.getCurrentGatewayAccountId(req)
 
   if (protoData) {
+    let paymentAmount = protoData.protoPaymentAmount * 100
```
Comment (kakumara) [inline app/controllers/make_a_demo_payment/confirm_controller.js]: We should have a test for this

File: app/controllers/make_a_demo_payment/confirm_controller.js [hunk 3]
```diff
@@ -15,6 +15,12 @@ module.exports = (req, res) => {
...
+    let index = protoData.protoPaymentAmount.indexOf('.')
+    if (index !== -1) {
```
Comment (kakumara) [inline app/controllers/make_a_demo_payment/confirm_controller.js]: Once again, this smells of trouble.

**Comment by SandorArpa** [comment]
Fix will be included in a different PR with relevant tests

**Pull request closed without merging**